quix-frontend/client/src/services/hooks/use-outside-alerter.ts

```diff
@@ -2,28 +2,27 @@ import { useEffect } from 'react';
 
 export const useOutsideAlerter = (refs: React.MutableRefObject<any>[], onClickOutside:() => void) => {
   useEffect(() => {
-      /**
-       * Alert if clicked on outside of element
-       */
-      const handleClickOutside = (event: any) => {
-        const isOutside = [];
-        refs.forEach(ref => {
-            if (ref.current && !ref.current.contains(event.target)) {
-                isOutside.push(true);
-            } else {
-                isOutside.push(false);
-            }
-        });
-        if (isOutside.every(i => i)) {
-            onClickOutside();
+    const handler = (event: any) => {
+      const inside = [];
+      refs.filter(ref => ref.current).forEach(ref => {
+        const parent = $(ref.current.parentElement);
+        const target = $(event.target);
+
+        if (!target.closest(parent).length || target.closest(event).length) {
+          inside.push(false);
+        } else {
+          inside.push(true);
         }
+      });
+
+      if (inside.every(i => !i)) {
+        onClickOutside();
       }
+    }
 
-      // Bind the event listener
-      document.addEventListener("mousedown", handleClickOutside);
-      return () => {
-          // Unbind the event listener on clean up
-          document.removeEventListener("mousedown", handleClickOutside);
-      };
+    document.addEventListener("mousedown", handler);
+    return () => {
+      document.removeEventListener("mousedown", handler);
+    };
   }, [refs]);
```

Comment:
what happens when the refs change during the component lifecycle?
who's make sure to remove the listeners from the document?

Reply:
the only time listeners are removed is when the component is destroyed.
this whole setup feels like a big memory leak?

Reply:
This function wrapped with `useEffect`, which means that every time that a ref changed, the cleanup run